torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -472,3 +472,93 @@ def place(vp: Placement, ip: Placement) -> Placement:
             ],
         )
         return result
+
+
+@register_prop_rule("aten.cat.default")
+def cat_rule(op_schema: OpSchema) -> OutputSharding:
+    dim = 0  # default dim = 0
+    tensor_list_specs = cast(List[DTensorSpec], op_schema.args_schema[0])
+    if (len(op_schema.args_schema) > 1):
+        dim = cast(int, op_schema.args_schema[1])
+    # normalize arguments
+    if dim < 0:
+        dim += tensor_list_specs[0].ndim
+
+    # check concat dim
+    needs_reshard_on_cat_dim = False
+    for spec in tensor_list_specs:
+        if dim < len(spec.placements) and spec.placements[dim].is_shard():
+            needs_reshard_on_cat_dim = True
+            spec.placements = unshard_tensor_dim(spec.placements, dim=dim)
+    if needs_reshard_on_cat_dim:
+        args_schema = (tensor_list_specs,) + op_schema.args_schema[1:]
+        suggested_schema = OpSchema(
+            func_schema=op_schema.func_schema,
+            args_schema=args_schema,
+            kwargs_schema=op_schema.kwargs_schema,
+        )
+        return OutputSharding(
+            None,
+            schema_suggestions=[suggested_schema],
+            failed_reason="All tensors in concat must have no sharding on cat dim, need to reshard!",
+        )
+    alphabet = "abcdefghijklmnopqrstuvwxyz"
+    einop_equation = ""
+    for spec in tensor_list_specs:
+        einop_equation += alphabet[:spec.ndim]
+        einop_equation += ','
+    einop_equation = einop_equation[:-1] + "->" + alphabet[:tensor_list_specs[0].ndim]
+    output_sharding = einop_rule(
+        einop_equation,
+        OpSchema(
+            func_schema=op_schema.func_schema,
+            args_schema=tuple(tensor_list_specs),
+            kwargs_schema={},
+        ),
+        linearity=False
+    )
+
+    if output_sharding.output_spec is None:
+        if output_sharding.schema_suggestions is not None:
+            return _update_schema_suggestion_for_cat(
+                output_sharding,
+                op_schema,
+                dim,
+            )
+        else:
+            return OutputSharding(None)
```

Comment:
shouldn't you return the `output_sharding` directly instead of creating a new one?